### PR TITLE
Intercom - Fix Intercom integration

### PR DESCRIPTION
**Pull request by arnaudbesnier**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -42,49 +42,44 @@ exports.init = function(opts) {
...
+
+      var query = {};
+      query[opts.integrations.stripe.userField] = userField;
+
+      return customerModel
+        .findOne({ where: query })
+        .then(function (customer) {
+          if (!customer) { return null; }
+          return customer.toJSON();
+        });
```
Comment (arnaudbesnier) [inline index.js]: Code indentation only

File: index.js [hunk 2]
```diff
@@ -42,49 +42,44 @@ exports.init = function(opts) {
...
-        if (customerId) {
-          return userModel
+      if (customerId) {
+        return userModel
           .findById(customerId)
-          .lean()
-          .exec(function (err, customer) {
-            if (err) { return reject(err); }
-            if (!customer) { return reject(); }
-            resolve(customer);
```
Comment (arnaudbesnier) [inline index.js]: I don't understand this code. It seems to fails sometimes.

Comment (arnaudbesnier) [inline index.js]: Probably a bad copy-paste from Express-Mongoose

Comment (SeyZ) [inline index.js]: You're right

File: index.js [hunk 3]
```diff
@@ -42,49 +42,44 @@ exports.init = function(opts) {
...
+          .then(function (customer) {
+            return customer;
```
Comment (SeyZ) [inline index.js]: `return userModel.findById(customerId);`

**Pull request merged**